src/main/java/az/edu/turing/happyfamilyv1/CollectionFamilyDao.java

```diff
@@ -0,0 +1,125 @@
+package az.edu.turing.happyfamilyv1;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CollectionFamilyDao implements FamilyDao {
+
+    private List<Family> families = new ArrayList<>();
+
+    @Override
+    public List<Family> getAllFamilies() {
+        return families;
+    }
+
+    @Override
+    public Family getFamilyById(int index) {
+        if (index >= 0 && index < families.size()) {
+            return families.get(index);
+        }
+        return null;
+    }
+
+    @Override
+    public boolean deleteFamily(int index) {
+        if (index >= 0 && index < families.size()) {
+            families.remove(index);
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean deleteFamily(Family family) {
+        return families.remove(family);
+    }
+
+    @Override
+    public void saveFamily(Family family) {
+        if (!families.contains(family)) {
```

Comment:
I think no need this check, we could get this information with indexOf which you already use in else block. Pls refactor the code